app/models/normalize_timdex.rb

```diff
@@ -33,10 +33,36 @@ def extract_results(results, norm)
 
   def extract_title(record)
     title = record['title']
-    title << (" (#{record['identifier'].to_s.gsub('MIT:archivespace:', '')})") if record['identifier'].present?
+    title << extract_collection_id(record['identifiers']) if record['identifiers'].present?
     title
   end
 
+  def extract_dates(dates)
+    # It is unlikely for a record to have more than one creation date, but just in case...
+    relevant_dates = dates.select { |date| date['kind'] == 'creation' }
+
+    # If the record *does* have more than one creation date, it's probably not worth determining which to display.
+    return if relevant_dates.count > 1
+
+    relevant_date = relevant_dates.first
+
+    # We are only concerned with creation dates that are ranges, since we harvest ASpace metadata at the collection
+    # level.
+    return unless relevant_date['kind'] == 'creation' && relevant_date['range'].present?
+
+    "#{relevant_date['range']['gte']}-#{relevant_date['range']['lte']}"
+  end
+
+  def extract_collection_id(identifiers)
+    relevant_ids = identifiers.map { |id| id['value'] if id['kind'] == 'Collection Identifier' }.compact
+
+    # In the highly unlikely event that there is more than one collection identifier, there's something weird going
+    # on with the record and we should skip it.
+    return if relevant_ids.count > 1
```

Comment:
This is kinda funny to me. I agree with it, but I wish I trusted we'd never need this line lol

Reply:
Hope for the best, plan for the worst, right? That said, I'm open to removing this and the comparable guard clause in the date method. This is the kind of thing I'd always do when handling Alma data, but I know less about the state of our ASpace data.